site/_posts/2022-06-22-what-is-scim.md

```diff
@@ -0,0 +1,147 @@
+
+There are additional [use cases outlined in the RFC](https://datatracker.ietf.org/doc/html/rfc7642#section-3).
+
+Let's dive into some of the major pieces of SCIM.
+
+## Client and Server
+
+There are two players in the SCIM game: the client and the server. The client provides the user data to the server.
+
+In the example above, when a new developer is added to an HR system, SCIM adds them to other systems too. In this case, the HR system is the *client* and the other systems are *servers*.
+
+{% plantuml source: _diagrams/blogs/scim/scim-new-developer.plantuml, alt: "Adding a new developer to all required systems." %}
+
+When you are evaluating SCIM solutions, make sure you understand if they implement SCIM as a client, a server, or both. For example, at this time, FusionAuth only implements SCIM as a server, capable of receiving user and group data.
+
+## Users and Groups
+
+Everything managed by SCIM is a resource. 
+
+There are three core resources: `Users`, `Groups` and `EnterpriseUsers`. SCIM can be extended to support other types of resources. These can be refinements of the core resources to model particular kinds of users. If you are using SCIM to provision users at a university you might have `Professors` and `Students` as resources. They can also model other kinds of identity management concepts, such as `Permissions`.
+
+`Users` and `EnterpriseUsers` are similar, with the `EnterpriseUsers` schema extending `Users`. Therefore, for the rest of this post, I'm going to focus on `Users`.
+
+`Users` have a defined set of attributes, many of which you might expect. These include `userName`, `name`, `userType` and more. Exactly what you'd expect a user to have. You can also put a user in a group.
+
+`Users` can also have emails, phone numbers, addresses and more. They can even have a list of x.509 certificates. Each user also has a password, which is write only.
+
+The required fields are few: the `userName` field is required. In addition, an `id` field must be created and assigned by the server.
+
+I'll let you review the [relevant section of the RFC](https://datatracker.ietf.org/doc/html/rfc7643#section-4.1) for the full list of attributes for this resource. 
+
+`Groups`, on the other hand, are much simpler. They have a required `displayName` and a list of members, as well as an `id`.
+
+### Schemas
+
+Schemas are a critical part of SCIM, and control what attributes each resource type has.
+
+The `Users`, `EnterpriseUsers` and `Groups` resources discussed above have their own standardized schema, but you can create your own or, as mentioned above, extend them.
+
+Schemas support single value attributes like `name` and `userType` with a variety of data types, as well as multi-valued attributes like phone numbers or emails.
+
+### Identifiers
+
+There are up to two identifiers for every resource in SCIM.
+
```

Comment:
Maybe add a mention of the $ref attribute that is returned from the SCIM server. That is very useful because it is the URI that can be used to retrieve data about that resource later. I believe all endpoints will return the $ref for the resource.